alma/controllers/hook/PaymentOptionsHookController.php

```diff
@@ -88,85 +97,98 @@ public function run($params)
-                            'merchantId' => Settings::getMerchantId(),
-                            'first' => $first,
-                            'creditInfo' => $creditInfo,
-                        ]);
-
-                        $template = $this->context->smarty->fetch(
-                            "module:{$this->module->name}/views/templates/hook/payment_button_pnx.tpl"
+            if ($isInstallmentAccordingToDeferred) {
+                $paymentOption = $this->createPaymentOption(
+                    $forEUComplianceModule,
+                    $textPaymentButton,
+                    $this->context->link->getModuleLink(
+                        $this->module->name,
+                        'payment',
+                        ['key' => $key],
+                        true
+                    ),
+                    $isDeferred,
+                    $valueBNPL
+                );
+                if (!$forEUComplianceModule) {
+                    $templateVar = [
+                        'desc' => $descPaymentButton,
+                        'plans' => (array) $plans,
+                        'deferred' => $feePlans->$key->deferred,
+                        'apiMode' => Settings::getActiveMode(),
+                        'merchantId' => Settings::getMerchantId(),
+                        'first' => $first,
+                        'creditInfo' => $creditInfo,
+                    ];
+                    if ($isDeferred) {
+                        $templateVar['installmentText'] = sprintf(
+                            $this->module->l('0 € today then %1$s on %2$s', 'PaymentOptionsHookController'),
+                            almaFormatPrice($plans[0]['purchase_amount'] + $plans[0]['customer_fee']),
+                            getDateFormat($locale, $plans[0]['due_date'])
                         );
-
-                        $paymentOptionPnx->setAdditionalInformation($template);
                     }
-                    $sortOptions[$key] = $feePlans->$key->order;
-                    $paymentOptions[$key] = $paymentOptionPnx;
+                    $this->context->smarty->assign($templateVar);
+                    $template = $this->context->smarty->fetch(
+                        "module:{$this->module->name}/views/templates/hook/{$fileTemplate}"
+                    );
+                    $paymentOption->setAdditionalInformation($template);
                 }
+                $sortOptions[$key] = $feePlans->$key->order;
+                $paymentOptions[$key] = $paymentOption;
             }
         }
 
         asort($sortOptions);
         $payment = [];
-        foreach ($sortOptions as $key => $option) {
+        foreach ($sortOptions as $key => $_) {
             $payment[] = $paymentOptions[$key];
         }
 
         return $payment;
     }
 
+    /**
+     * Create Payment option
+     *
+     * @param bool $forEUComplianceModule
+     * @param string $ctaText
+     * @param string $action
+     * @param bool $deferred
```

Comment:
Please name this variable `isDeferred` to avoid confusions about deferred concepts 😉

Reply:
Done